### PR TITLE
*: add release notes for v3.0.5

### PR DESCRIPTION
**Pull request by zimulala**
Add the changelog for v3.0.5

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @bb7133 @lonng

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,50 @@
 # TiDB Changelog
 All notable changes to this project will be documented in this file. See also [Release Notes](https://github.com/pingcap/docs/blob/master/releases/rn.md), [TiKV Changelog](https://github.com/tikv/tikv/blob/master/CHANGELOG.md) and [PD Changelog](https://github.com/pingcap/pd/blob/master/CHANGELOG.md).
 
+## [3.0.5] 2019-10-23
+## SQL Optimizer
```
Comment (ericsyh) [inline CHANGELOG.md]:
```suggestion
### SQL Optimizer
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,50 @@
...
+* Fix the issue that `IndexJoin` on the partition table returns incorrect results [#12712](https://github.com/pingcap/tidb/pull/12712)
+* Fix the issue that the `ifnull` function on the top of the outer join `Apply` operator returns incorrect results [#12694](https://github.com/pingcap/tidb/pull/12694)
+* Fix the issue of update failure when a subquery was included in the `where` condition of `UPDATE` [#12597](https://github.com/pingcap/tidb/pull/12597)
+* Fix the issue that outer join was incorrectly converted to inner join when the `cast` function was included in the query conditions [#12790](https://github.com/pingcap/tidb/pull/12790)
+* Fix incorrect expression passing in the join condition of `AntiSemiJoin` [#12799](https://github.com/pingcap/tidb/pull/12799)
+* Fix the statistics error caused by shallow copy when initializing statistics [#12817](https://github.com/pingcap/tidb/pull/12817)
+* Fix the issue that the `str_to_date` function in TiDB returns a different result from MySQL when the date string and the format string do not match [#12725](https://github.com/pingcap/tidb/pull/12725)
+
+## SQL Execution Engine
```
Comment (ericsyh) [inline CHANGELOG.md]:
```suggestion
### SQL Execution Engine
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,50 @@
...
+* Fix the `invalid list index` error reported when canceling DDL jobs [#12671](https://github.com/pingcap/tidb/pull/12671)
+* Fix the issue that arrays were out of bounds when Window Functions are used [#12660](https://github.com/pingcap/tidb/pull/12660)
+* Improve the behavior of the `AutoIncrement` column when it is implicitly allocated, to keep it consistent with the default mode of MySQL auto-increment locking (["consecutive" lock mode](https://dev.mysql.com/doc/refman/5.7/en/innodb-auto-increment-handling.html)): for the implicit allocation of multiple `AutoIncrement` IDs in a single-line `Insert` statement, TiDB guarantees the continuity of the allocated values. This improvement ensures that the JDBC `getGeneratedKeys()` method will get the correct results in any scenario. [#12602](https://github.com/pingcap/tidb/pull/12602)
+* Fix the issue that the query is hanged when `HashAgg` serves as a child node of `Apply` [#12766](https://github.com/pingcap/tidb/pull/12766)
+* Fix the issue that the `AND` and `OR` logical expressions return incorrect results when it comes to type conversion [#12811](https://github.com/pingcap/tidb/pull/12811)
+
+## Server
```
Comment (ericsyh) [inline CHANGELOG.md]:
```suggestion
### Server
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,50 @@
...
+* Support printing the execution plan compression-encoded information in the slow log. This feature is enabled by default and can be controlled by using the `slow-log-plan` configuration or the `tidb_record_plan_in_slow_log` variable. In addition, the `tidb_decode_plan` function can parse the execution plan column encoding information in the slow log into execution plan information. [#12808](https://github.com/pingcap/tidb/pull/12808)
+* Support displaying memory usage information in the `information_schema.processlist` table [#12801](https://github.com/pingcap/tidb/pull/12801)
+* Fix the issue that an error and an unexpected alarm might occur when the TiKV Client judges an idle connection [#12846](https://github.com/pingcap/tidb/pull/12846)
+* Fix the issue that the `INSERT IGNORE` statement performance is decreased because `tikvSnapshot` does not properly cache the KV results of `BatchGet()` [#12872](https://github.com/pingcap/tidb/pull/12872)
+* Fix the issue that the TiDB response speed was relatively low because of slow connection to some KV services [#12814](https://github.com/pingcap/tidb/pull/12814)
+
+## DDL
```
Comment (ericsyh) [inline CHANGELOG.md]:
```suggestion
### DDL
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,50 @@
...
+* Support multiple `unique`s when creating a unique index using the `Create Table` statement [#12463](https://github.com/pingcap/tidb/pull/12463)
+* Fix the issue that populating the default value of this column for existing rows might cause an error when adding a Bit type column using `Alter Table` [#12489](https://github.com/pingcap/tidb/pull/12489)
+* Fix the failure of adding a partition when the Range partitioned table uses a Date or Datetime type column as the partitioning key [#12815](https://github.com/pingcap/tidb/pull/12815)
+* Support checking the consistency of the partition type and the partition key type when creating a table or adding a partition, for the Range partitioned table with the Date or Datetime type column as the partition key [#12792](https://github.com/pingcap/tidb/pull/12792)
+* Add a check that the Unique Key column set needs to be greater than or equal to the partitioned column set when creating a Range partitioned table [#12718](https://github.com/pingcap/tidb/pull/12718)
+
+## Monitor
```
Comment (ericsyh) [inline CHANGELOG.md]:
```suggestion
### Monitor
```

**Pull request merged**